src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -819,32 +821,48 @@ class spell_warr_sword_and_board : public SpellScriptLoader
         }
 };
 
-// 32216 - Victorious
-// 82368 - Victorious
-class spell_warr_victorious : public SpellScriptLoader
+// 34428 - Victory Rush
+class spell_warr_victory_rush : public SpellScriptLoader
 {
     public:
-        spell_warr_victorious() : SpellScriptLoader("spell_warr_victorious") { }
+        spell_warr_victory_rush() : SpellScriptLoader("spell_warr_victory_rush") { }
 
-        class spell_warr_victorious_AuraScript : public AuraScript
+        class spell_warr_victory_rush_SpellScript : public SpellScript
         {
-            PrepareAuraScript(spell_warr_victorious_AuraScript);
+            PrepareSpellScript(spell_warr_victory_rush_SpellScript);
 
-            void HandleEffectProc(AuraEffect const* /*aurEff*/, ProcEventInfo& /*eventInfo*/)
+            bool Validate(SpellInfo const* /*spellInfo*/) override
             {
-                PreventDefaultAction();
-                GetTarget()->RemoveAura(GetId());
+                return ValidateSpellInfo
+                ({
+                    SPELL_WARRIOR_VICTORIOUS,
+                    SPELL_WARRIOR_VICTORY_RUSH_HEAL
+                });
+            }
+
+            void HandleOnHit()
+            {
+                if (!GetHitUnit())
+                    return;
+
+                Unit* caster = GetCaster();
+
+                if (caster->HasAura(SPELL_WARRIOR_VICTORIOUS))
```

Comment:
This check is not needed, since Victory Rush can only be used when he has that aura anyways.

CasterAuraState = 10 (AURA_STATE_WARRIOR_VICTORY_RUSH)